src/libmongoc/src/mongoc/mongoc-client.c

```diff
@@ -1084,54 +1063,53 @@ mongoc_client_set_ssl_opts (mongoc_client_t *client,
 #endif
 
 
-/*
- *--------------------------------------------------------------------------
- *
- * mongoc_client_new_from_uri --
- *
- *       Create a new mongoc_client_t for a mongoc_uri_t.
- *
- * Returns:
- *       A newly allocated mongoc_client_t.
- *
- * Side effects:
- *       None.
- *
- *--------------------------------------------------------------------------
- */
-
 mongoc_client_t *
 mongoc_client_new_from_uri (const mongoc_uri_t *uri)
 {
+   mongoc_client_t *client;
+   bson_error_t error = {0};
+
+   if (!(client = mongoc_client_new_from_uri_with_error (uri, &error))) {
+      MONGOC_ERROR ("%s", error.message);
+   }
+
+   return client;
+}
+
+
+mongoc_client_t *
+mongoc_client_new_from_uri_with_error (const mongoc_uri_t *uri,
+                                       bson_error_t *error)
+{
+   mongoc_client_t *client;
    mongoc_topology_t *topology;
 
    topology = mongoc_topology_new (uri, true);
 
+   if (!topology->valid) {
+      if (error) {
+         memcpy (error, &topology->scanner->error, sizeof (bson_error_t));
+      }
+
+      mongoc_topology_destroy (topology);
+
+      return NULL;
```

Comment:
Returning `NULL` on failure to apply SRV records looks like the cause of `/initial_dns_seedlist_discovery/null_error_pointer` test failures in test-mongoc-dns.c. [Evergreen link](https://evergreen.mongodb.com/lobster/evergreen/task/mongo_c_driver_asan_ubuntu_debug_compile_asan_clang_openssl_patch_a203bafab7d412ed05f33b99229bacbc043c2268_61cb42c1d1fe0720f5f9a13f_21_12_28_17_00_50/0/task#bookmarks=0%2C11782&l=1&shareLine=11769).

I suggest removing the ``/initial_dns_seedlist_discovery/null_error_pointer` test. It tests calls on an invalid topology. But an invalid topology is no longer accessible to a user.

Reply:
That test and another that were relying on returning a client/pool with an invalid topology have been removed. Feel free to resolve.